commit 3118ea28d65b8bd95a91e7d9a3a74cd5f22f0fe3
Author: Alexander K <ak@natsys-lab.com>
Date:   Mon Mar 28 00:08:35 2016 +0300

    Remove msg check from netlink_sendmsg() which is never satisfied.
    Modern kernels remove NETLINK_MMAP functionality at all,
    while 4.1 just bearks it.

diff --git a/net/netlink/af_netlink.c b/net/netlink/af_netlink.c
index d139c43a..1e4da4e9 100644
--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -2367,7 +2367,6 @@ static int netlink_sendmsg(struct socket *sock, struct msghdr *msg, size_t len)
 	 * sendmsg(), but that's what we've got...
 	 */
 	if (netlink_tx_is_mmaped(sk) &&
-	    msg->msg_iter.type == ITER_IOVEC &&
 	    msg->msg_iter.nr_segs == 1 &&
 	    msg->msg_iter.iov->iov_base == NULL) {
 		err = netlink_mmap_sendmsg(sk, msg, dst_portid, dst_group,
